cockatrice/src/client/ui/picture_loader/picture_loader_local.cpp

```diff
@@ -59,84 +56,89 @@ QImage PictureLoaderLocal::tryLoad(const CardInfoPtr &toLoad) const
-    }
+    // Most-to-least specific, these will fall through in order.
 
+    QStringList nameVariants;
+    // cardName_providerId
+    if (!providerId.isEmpty()) {
+        nameVariants << QString("%1-%2").arg(correctedCardName, providerId);
+        nameVariants << QString("%1_%2").arg(correctedCardName, providerId);
+    }
+    // cardName_setName_collectorNumber & setName-collectorNumber-cardName
+    if (!setName.isEmpty() && !collectorNumber.isEmpty()) {
+        nameVariants << QString("%1_%2_%3").arg(correctedCardName, setName, collectorNumber)
+                     << QString("%1-%2-%3").arg(setName, collectorNumber, correctedCardName);
+    }
+    // cardName_setName
     if (!setName.isEmpty()) {
-        picsPaths << picsPath + "/" + setName + "/" + correctedCardName
-                  // We no longer store downloaded images there, but don't just ignore
-                  // stuff that old versions have put there.
-                  << picsPath + "/downloadedPics/" + setName + "/" + correctedCardName;
+        nameVariants << QString("%1_%2").arg(correctedCardName, setName)
+                     << QString("%1-%2").arg(setName, correctedCardName);
     }
+    // cardName
+    nameVariants << correctedCardName;
 
-    // Iterates through the list of paths, searching for images with the desired
-    // name with any QImageReader-supported extension
-    for (const auto &_picsPath : picsPaths) {
-        if (!QFileInfo(_picsPath).isFile()) {
+    for (const QString &nameVariant : nameVariants) {
+        if (nameVariant.isEmpty()) {
             continue;
         }
 
-        imgReader.setFileName(_picsPath);
-        if (imgReader.read(&image)) {
-            qCDebug(PictureLoaderLocalLog).nospace()
-                << "[card: " << correctedCardName << " set: " << setName << "]: Picture found on disk.";
-            return image;
+        QStringList candidatePaths;
+        const auto matches = customFolderIndex.values(nameVariant);
+
+        for (const QString &path : matches) {
+            candidatePaths << path;
         }
-        imgReader.setFileName(_picsPath + ".full");
-        if (imgReader.read(&image)) {
-            qCDebug(PictureLoaderLocalLog).nospace()
-                << "[card: " << correctedCardName << " set: " << setName << "]: Picture.full found on disk.";
-            return image;
+
+        if (!setName.isEmpty()) {
+            candidatePaths << picsPath + "/" + setName + "/" + nameVariant;
+            candidatePaths << picsPath + "/downloadedPics/" + setName + "/" + nameVariant;
         }
-        imgReader.setFileName(_picsPath + ".xlhq");
-        if (imgReader.read(&image)) {
-            qCDebug(PictureLoaderLocalLog).nospace()
-                << "[card: " << correctedCardName << " set: " << setName << "]: Picture.xlhq found on disk.";
-            return image;
+
+        for (const QString &path : candidatePaths) {
+            for (const QString suffix : {"", ".full", ".xlhq"}) {
```

Comment:
I know that the previous code used the full and xlhq suffixes (these are mws holdovers to make copying images from there possible afaik) however our code should ignore all suffixes anyway, it already reads images solely based on contents

note that there are some cards with dots in them, these still need to be possible to read